### ISSUE TITLE
"Move to applications folder" notice keeps popping up, updates fail

### ISSUE THREAD

**Issue by gkertai** [videovillage/Screen-Issues#17]
Upon startup, the Screen app kept showing the notification "Move to Applications Folder?" even though it was already in the /Applications folder.

![Screen Shot 2020-01-28 at 13 41 27](https://user-images.githubusercontent.com/1874498/73265841-00c5ef00-41d6-11ea-9621-f019bfaa7286.PNG)

I clicked yes, and it moved itself into the Trash in two copies: *Screen.app* and *Screen 13.41.10.app*. I restored the former, and disabled the notification by the checkbox.

But now when I check for updates it says "Update Error" because it thinks it's still not in the Applications folder.

![Screen Shot 2020-01-28 at 13 55 34](https://user-images.githubusercontent.com/1874498/73266021-58645a80-41d6-11ea-9fc7-23b21e99e5c6.PNG)

Well, it *is* in the /Applications folder…

```bash
[GMacBk:~] mdfind -name Screen.app
/Applications/Screen.app
```

I did some digging, and apparently when I start the app, it copies itself into `$TMPDIR/AppTranslocation/…random…GUID…/d/Screen.app`. Maybe this helps. However the problem persists even if I remove the copy from $TMPDIR.

```bash
[GMacBk:~] find / -name 'Screen.app' 2>/dev/null
/private/var/folders/mp/t…n/T/AppTranslocation/BDBDD27E-50C1-4980-9240-3DE7D754D505/d/Screen.app
/Applications/Screen.app
```

MacOS High Sierra 10.13.6

**Comment by gregcotten** [comment]
Hi Gabor! This is a first! The technology behind “move to applications” and update checking are completely different, so if both are reporting the app isn’t in the Applications folder I would agree that something is wrong with your setup. Normally if a non-Mac App Store .app is opened from outside /Applications, the app is actually moved to a random temporary location (trans located) for security reasons. This leads me to believe the .app you are opening is not actually in /Applications. 

If you are opening Screen via a shortcut in the dock, option click it in the dock and click Show in Finder to see where the .app is located - it’s possible you’re opening a duplicate of Screen from elsewhere (outside /Applications). Or if you’re opening via spotlight, hitting command-enter instead of enter should take you to the location of the .app.

Another option is to search for and remove Screen.app from everywhere you can find it, and then re-download from your license page and manually move to the /Applications directory, then open from there.

**Comment by gkertai** [comment]
Thanks for your quick reply Greg!

I tried opening it from Spotlight, Launchpad, and directly from the /Applications folder in Finder. In every case it's being run from the temp copy.

```bash
[GMacBk:~] ps -A -o comm | grep Screen.app
/private/var/folders/mp/t…n/T/AppTranslocation/930FB276-915D-4359-8AC0-5104445619B5/d/Screen.app/Contents/MacOS/Screen
```

Deleting then reinstalling directly to Applications did not help first. Then I deleted again, extracted to Downloads, ran from Downloads (it did not complain this time), closed it, moved it manually to Applications, ran it from there, and voila… update checks work fine, and it's no longer being run from temp.

```bash
[GMacBk:~] ps -A -o comm | grep Screen.app
/Applications/Screen.app/Contents/MacOS/Screen
```

I've never had this happen to me with any other app. I guess we can close this ticket now, it may or may not be related to the Screen app. I'll leave this here for future reference, in case anybody else struggles with this.
